src/execution/execute.js

```diff
@@ -780,9 +823,17 @@ function completeListValue(
   // where the list contains no Promises by avoiding creating another Promise.
   const itemType = returnType.ofType;
   let containsPromise = false;
-  const completedResults = result.map(item => {
-    const completedItem =
-      completeValueCatchingError(exeContext, itemType, fieldASTs, info, item);
+  const completedResults = result.map((item, index) => {
+    const childExePath = exePath.slice();
+    childExePath.push(index);
+
+    const childInfo = Object.assign({}, info, {
```

Comment:
I think it is necessary, since each child executor has a different `info` object with a different `exePath` property. Since the execution can run concurrently in async, the child executors can't share the same `info` object.

Does it seem fair? Another option that avoids copying objects around is to have a tree-like data structure and require anyone who wants to collect the `path` to walk the tree up to the root. Which complicates the interface and the implementation and only saves some amount of object allocations.